### ISSUE TITLE
dwrite_2.h Build

### ISSUE THREAD

**Issue by ekicyou** [sharpdx/SharpDX#526]
I want to use "dwrite_2.h" API. (windows8.1 add)
But I don't know how is the good edit for the "DirectWrite/Mapping.xml".
https://github.com/sharpdx/SharpDX/blob/master/Source/SharpDX.Direct2D1/DirectWrite/Mapping.xml

Please tell me how.

**Comment by xoofx** [comment]
Which assemblies are you using? Windows 8.1 API should be mostly available from DirectX11_2 assemblies. Though, haven't checked how much changed in Windows 8.1.

**Comment by ekicyou** [comment]
I am using SharpDX.Direct2D1 2.6.3.
windows8.1 now supports Japanese vertical text, but in relation to the additional API.

  1. specified vertical in the TextFormat.
    * format.ReadingDirection = DWRITE_READING_DIRECTION_TOP_TO_BOTTOM
    * format.FlowDirection = DWRITE_FLOW_DIRECTION_RIGHT_TO_LEFT
  2. "IDWriteTextLayout::draw" has been run, giving the IDWriteTextRenderer method.
  3. I have failed in 0x8898500A (DWRITE_E_TEXTRENDERERINCOMPATIBLE) response.

If the specified vertical in the TextFormat "IDWriteTextLayout::draw" the renderer passed as an argument is [IDWriteTextRenderer], not [IDWriteTextRenderer1] interface is required.

To support the IDWriteTextRenderer1 in 2.6.3 definition does not exist. Therefore verified or cannot modify the source code, but the mapping.xml specifications are unknown and was abandoned.

**Comment by xoofx** [comment]
Indeed, It seems that dwrite_2.h is not included in mapping.xml, so we will have to add support for it. Though, I have currently no dedicated spare time for working on this.

**Comment by ekicyou** [comment]
Thank you! Wait for support.

**Comment by huan086** [comment]
@ekicyou Check out my fork at https://github.com/huan086/SharpDX/commits/vs2015.
I've added the mappings for IDWriteTextRenderer1.

The interfaces required for coloured emojis definitely work. Haven't tested the vertical text

**Comment by snowcrack** [comment]
I struggle with the same problem. Can't set ReadingDirection to TopToBottom.

> 'HRESULT: [0x8898500A], Module: [Unknown], ApiCode: [Unknown/Unknown], Message: The version of the text renderer interface is not compatible.'

Are there any updates? Or is there a workaround?

**Comment by xoofx** [comment]
@snowcrack with little details, it is hard to say. Latest version should have dwrite_2...etc.  but can't tell if they are exposed correctly. `IDWriteTextRenderer1` doesn't seem to be exposed, so if you want this, PR welcome.

**Comment by snowcrack** [comment]
When I set the FlowDirection to LeftToRight and the ReadingDirection  to TopToBottom, it is the TextLayout.Draw method, which raise the exception.
With the horizontal render directions, all works fine.

I use my own TextRenderer. This also inherits from TextRenderer1.

Unfortunately, I have no idea how to fix this in sharpDX.

**Comment by xoofx** [comment]
afaik TextRenderer1 is not exposed in SharpDX so far.
